jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -741,6 +741,18 @@ else if (_decryptedInput == null)
                                         throw new IllegalStateException("Unexpected unwrap result " + unwrapResultStatus);
 
                                     case BUFFER_UNDERFLOW:
+                                        // Continue if we can compact?
+                                        if (BufferUtil.compact(_encryptedInput))
+                                            continue;
+
+                                        // Are we out of space?
+                                        if (BufferUtil.space(_encryptedInput) == 0)
+                                        {
+                                            BufferUtil.clear(_encryptedInput);
+                                            throw new SSLHandshakeException("Encrypted buffer max length exceeded");
+                                        }
+
```

Comment:
I suspect we should also have:
```java
                                    // if we just filled some
                                    if (netFilled > 0)
                                        continue; // try filling some more
```

Reply:
This is handled up at line 700 already.
And it should be `break decryption;` to force a read/fill again on 9.3.x